pocs/serial_handlers/protocol_arduinosimulator.py

```diff
@@ -173,17 +178,15 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.logger = pocs.utils.logger.get_root_logger()
         self.simulator_thread = None
-        # The elements of these queues are of type bytes. This means we aren't fully controlling
-        # the baudrate, but that should be OK.
-        self.relay_queue = queue.Queue(maxsize=100)
-        self.json_queue = queue.Queue(maxsize=10000)
+        self.relay_queue = queue.Queue(maxsize=1)
+        self.json_queue = queue.Queue(maxsize=1)
         self.json_bytes = bytearray()
         self.stop = threading.Event()
         self.stop.set()
         self.device_simulator = None
 
     def __del__(self):
-        if self.simulator_thread:
+        if self.simulator_thread:  # pragma: no cover
```

Comment:
This seems like a lot to not cover

Reply:
It basically is covering a misuse of the simulator, i.e. of test code.